src/test/java/com/simon/smile/user/address/AddressServiceTest.java

```diff
@@ -0,0 +1,180 @@
+        assertThat(foundAddress.getAddress()).isEqualTo(address.getAddress());
+        assertThat(foundAddress.getPhone()).isEqualTo(address.getPhone());
+        assertThat(foundAddress.getOwner()).isEqualTo(address.getOwner());
+        assertThat(foundAddress.isDefault()).isEqualTo(address.isDefault());
+        verify(addressRepository, times(1)).findById(anyInt());
+    }
+
+    @Test
+    @DisplayName("Verify find addresses by owner Id success")
+    void validFindAddressesByOwnerIDSuccess() {
+        given(addressRepository.findByOwnerId(anyInt())).willReturn(addressList);
+
+        List<Address> foundAddresses = addressService.findByOwnerId(appUser.getId());
+
+        assertThat(foundAddresses).hasSize(2);
+        assertThat(foundAddresses.get(0).getAddress()).isEqualTo(address.getAddress());
+        verify(addressRepository, times(1)).findByOwnerId(anyInt());
+    }
+
+    @Test
+    @DisplayName("Verify update address success")
+    void validUpdateSuccess() {
+        var newAddress = new Address()
+                .setId(1)
+                .setAddress("new address")
+                .setPhone("new phone")
+                .setOwner(appUser)
+                .setDefault(true);
+        given(addressRepository.findById(anyInt())).willReturn(Optional.of(address));
+        given(addressRepository.save(any(Address.class))).willReturn(newAddress);
+
+        Address updatedAddress = addressService.update(1, newAddress);
+        assertThat(updatedAddress.getAddress()).isEqualTo(newAddress.getAddress());
+        assertThat(updatedAddress.getPhone()).isEqualTo(newAddress.getPhone());
+        assertThat(updatedAddress.getOwner()).isEqualTo(newAddress.getOwner());
+        assertThat(updatedAddress.isDefault()).isEqualTo(newAddress.isDefault());
+        verify(addressRepository, times(1)).findById(anyInt());
+        verify(addressRepository, times(1)).save(any(Address.class));
+    }
+
+    @Test
+    @DisplayName("Verify update address error when the ID not exist")
+    void validUpdateErrorWhenIDNotExist() {
+        given(addressRepository.findById(anyInt()))
+                .willThrow(new ObjectNotFoundException(ERROR_ADDRESS_NOT_FOUND));
+
+        Throwable throwable = catchThrowable(() -> addressService.update(1, address));
+        assertThat(throwable).isInstanceOf(ObjectNotFoundException.class);
+        assertThat(throwable).hasMessage(ERROR_ADDRESS_NOT_FOUND);
+        verify(addressRepository, times(1)).findById(anyInt());
+        verify(addressRepository, times(0)).save(any(Address.class));
+    }
+
+    @Test
+    @DisplayName("Verify find address by ID error when id not exist")
+    void validFindByIdErrorWhenIDNotExist() {
+        given(addressRepository.findById(anyInt()))
+                .willThrow(new ObjectNotFoundException(ERROR_ADDRESS_NOT_FOUND));
+
+        Throwable throwable = catchThrowable(() -> addressService.findById(1));
+        assertThat(throwable).isInstanceOf(ObjectNotFoundException.class);
```

Comment:
(sonar-qube) Join these multiple assertions subject to one assertion chain.
---
Consecutive AssertJ `assertThat` statements should be chained.

Reply:
done